File: API/Entities/Scrobble/ScrobbleEventFilter.cs
namespace API.Entities.Scrobble;

public class ScrobbleEventFilter
{
    /// <summary>
    /// Which field to sort on
    /// </summary>
    public ScrobbleEventSortField Field { get; set; } = ScrobbleEventSortField.LastModified;

    /// <summary>
    /// If the sort should be a descending sort
    /// </summary>
    public bool IsDescending { get; set; } = true;
    /// <summary>
    /// A query to search against
    /// </summary>
    public string Query { get; set; }
    /// <summary>
    /// Include reviews in the result - Note: Review Scrobbling is disabled
    /// </summary>
    public bool IncludeReviews { get; set; } = false;
}

File: UI/Web/src/app/_models/scrobbling/scrobble-event-filter.ts
export enum ScrobbleEventSortField {
  None = 0,
  Created = 1,
  LastModified = 2,
  Type= 3,
  Series = 4,
  IsProcessed = 5
}

export interface ScrobbleEventFilter {
  field: ScrobbleEventSortField;
  isDescending: boolean;
  query?: string;
}
